Add CreateDebugNuget script

althurman opened the pull request:
Debug script to build local reactive domain nuget.

Note: The actual nugets are the resulting .nupkg files that will be in the tools folder. The version number is randomly generated. Once copying locally , the user must add the containing folder to the list of package sources in Visual Studio:

https://docs.microsoft.com/en-us/nuget/consume-packages/install-use-packages-visual-studio#package-sources


PR also adds gitignore file ignoring .nupkg files

althurman committed:
Add CreateDebugNuget script

- add gitignore file ignoring .nupkg files

joshkempner commented on tools/CreateDebugNuget.ps1:
```diff
@@ -0,0 +1,284 @@
+
+Write-Host ("Powershell script location is " + $PSScriptRoot)
+
+$RDMajor = Get-Random -Maximum 100
+$RDMinor = Get-Random -Maximum 100
+$RDRevision = Get-Random -Maximum 100
+$RDVersion = $RDMajor.ToString() + "."+ $RDMinor.ToString() + "." + $RDRevision.ToString()
```
This isn't right. The version should come from the buildprops file.

althurman replied:
It should come from the build props file on a real nuget build. For a local debug nuget, the version number doesnt matter does it? 

Remember: The developer is supposed to update the build.props to create the new official nuget, otherwise the travis build will will fail. In the debug case, using the number in build.props would create a duplicate of the current version that is in the PKI Private feed. So we would have to increment that number anyway. Instead of making calles to nuget.org to check against the version in build props, we might as well just generate a random number for the debug nuget

joshkempner replied:
I understand the desire to keep this simple, but we should at least make sure that it gives us a workable version. Perhaps the following would work:
1. Get major, minor, and revision from buildprops.
2. Append "-localNNN" to the version string where NNN is a random number.

The suffix will mark it as unstable and give you a different revision number each time (mod the vagaries of random number generators and chance).

althurman committed:
Address review comments

- Make random temp number 1000 - 10000
- Add random suffix to the end of existing version in build.props
